### ISSUE TITLE
How to integrate this videojs-vimeo plugin with pre-roll vast!? 

### ISSUE THREAD

**Issue by mina64** [videojs/videojs-vimeo#85]
Hello Everyone,
Could you please help me to integrate this plugin with Vast pre-roll (Video ad from DoubleClick).

Thanks

**Comment by glpunk** [comment]
Did you found any solution? same problem here. I tried to integrate vast plugin for videoJS but both together doesn't work.

**Comment by mina64** [comment]
Hey Glpunk,
Yes, I found solution for my problem finally after days investigation so I can share my code with you. please send email to phpcodeclinic@gmail.com, I will forward you zip file that contains my solution with source code :)

**Comment by mina64** [comment]
Hey Glpunk,
Yes, I found solution for my problem finally after days investigation so I can share my code with you. please send me your email address here.

**Comment by glpunk** [comment]
HI @mina64 thanks for your fast response.

My email is the same username here AT gmail 

Thank you!

**Comment by Malmoc** [comment]
@mina64 - Could you share your code with me too? My email is malmocdk AT gmail.com

**Comment by mina64** [comment]
I just sent you but note that we can not customize Vimeo Player that's why you need to use direct file link for vimeo video, you can find direct file link in your vimeo video settings.

**Comment by mina64** [comment]
Sure Malmoc I am going to send you now

**Comment by mina64** [comment]
malmoc@gmail.com not working

**Comment by mina64** [comment]
@glpunk glpunk Please confirm if you got the file. Thanks

**Comment by mina64** [comment]
malmocdk worked, please confirm if you got it.

**Comment by mina64** [comment]
Enjoy it!

**Comment by mina64** [comment]
I have to run so please if you send me any message, be patient! Thanks

**Comment by glpunk** [comment]
@mina64 Got the file, thank you! I will check soon.

**Comment by mina64** [comment]
Okay Cool, Please let me know if it solved your issue. Thanks :)

There is a file Ads.js that you can put your own video ad tag url, for test purpose, I used DFP default video tag link.

**Comment by glpunk** [comment]
HI @mina64 

The code is working fine thank you!, works with my vast server as well (is not doubleClick).

BUT, you're getting the video as mp4 directly from vimeo, I would Like to playback with m4s chunks as the vimeo plugin do, as vimeo works actually. 

I will try to use THIS ad-server code with the vimeo-plugin, maybe will work!

**Comment by mina64** [comment]
Correct, As I mentioned earlier, Vimeo doesn't allow anyone to modify its player so the only way to intgrate Vimeo videos with ad services, is to use Vimeo Video Direct File Links which is in mp4 format and put it in our player (in my case, video palyer is VideoJS).

VideoJS is very powerful as we can fully customize it.

Please let me know if you need my help.

**Comment by thanhtu90** [comment]
dear mina64 ,
I Had the same issues , could you please send me your solution too. Thank you a lot.

**Comment by mina64** [comment]
Sure, Just sent it! Please check your email :)

Have a great time!

**Comment by thanhtu90** [comment]
I got it, but I can't find the direct link from vimeo anymore, do you have solution to work around ?

**Comment by mina64** [comment]
Vimeo Direct Links in this file:
vimeovast/examples/index.php

**Comment by thanhtu90** [comment]
I mean vimeo link from my vimeo account on https://vimeo.com/user57411025/videos

**Comment by mina64** [comment]
Open that index.php mentioned in my earlier comment, you will find those links in line #33 and #34 :)

**Comment by mina64** [comment]
oh I got you, see dear, Vimeo doesn't allow you to customize their player or use 3rd party player, instead they provide you Video Direct Link that you can use them in 3rd party player. Like what I have done. so if you want to embed your vimeo videos, just login to your vimeo account and click on the desired videos and the go to Settings and then click on Video Files or Files and find Direct Links in mp4 format.

Please let me know if you still have any question.

**Comment by thanhtu90** [comment]
Thanks mate, but vimeo provide Video Direct Link for Pro User only. Anyway thank you for your help.

**Comment by marinamat76** [comment]
HI @mina64 ! 

Could you share your code with me too? 
My email is marinamat76 AT gmail.com

Thank you!

**Comment by mina64** [comment]
@marinamat76  Sure,  I am going to send now. Hold on please.

**Comment by mina64** [comment]
Sorry gmail is not allowing me to send that attachment, because it contains js file so I am going to send you through wetransfer.com, once you got it please confirm. Thanks

**Comment by marinamat76** [comment]
Thank you very much, 
but unfortunately I have'nt received anything by wetransfer.com

**Comment by mina64** [comment]
Please check your email again now and confirm! Thanks

**Comment by marinamat76** [comment]
Thank you,
now i have received!

Marina.

**Comment by dchristof** [comment]
@mina64 can you please share your code with me also (dchristof AT gmail.com? I am about to upgrade my vimeo account from PLUS to PRO just to be able to get the direct mp4 link of the video file so that I can use it on my site with the videojs vimeo plugin and vast ads. If any of you has a direct mp4 link please post it (or send it to me) so that I can test it to see if ads get displayed before I upgrade my account. Thanks so much in advance.

**Comment by gkatsev** [comment]
Master was just updated to support Video.js 5, this means that any of the VAST plugins for videojs should potentially work with it. I still think there might be some bugs that need to be ironed, but it's mostly there.

**Comment by lminarini** [comment]
@mina64 would you mind share the code with me too? My email is larissaminarini AT gmail.com

**Comment by jeter54** [comment]
@mina64   will you please share the code with me as well.  My email is adam.reader@me.com THANKS

**Comment by marzton** [comment]
@mina64 Would you please share the source code with me? I'm trying to integrate Vimeo/ DFP and am having some trouble. robertm@icdnet.com

**Comment by niralmodi** [comment]
can you please share code on niralmodi@gmail.com

**Comment by kgabalesvideomaker** [comment]
Hello, please share the code on katz@videomaker.com

**Comment by sthiruna28** [comment]
Hi, Can you please share the code. I am trying to integrate Vimeo with VAST and am having problems. Thanks. seannagari69 AT gmail.com

**Comment by praznovskymartin** [comment]
Hi there, I have the same issue. Could you please share the code. Thanks praznovsky.martin AT gmail.com

**Comment by sosubatomic** [comment]
@mina64 Would you mind sharing your code? michele AT sosubatomic.com

**Comment by ehosgor** [comment]
Hello, please share the code on engin.hosgor@gmail.com

**Comment by marzton** [comment]
Hello,

I no longer have access to that particular code. Ever since Covid, my
employer succumbed and sold the business. I was laid off Christmas 2020.

Achieving a successful implementation of a pre, post, mid roll ad greatly
relied on the particular DFP line item ad set up. They have a VAST test
which I utilized for set up and debugging. As far as targeting a specific
video, I fed Google many parameters: query string, WP ID, post title, and
sponsored custom field checkbox. All really to make ad ops coordinator work
easier as well as to prevent any competitive advertisers from displaying on
the video post.

They may have updated since I last coded a working version. However the set
up was simple. First give your HTML5 player a unique #ID. Add and fill the
video-JS JSON option parameters, player #ID. Test first with a Google pre
roll VAST url. It will greatly help was it has the query parameters
required.

I believe there was one extra JS add-on to Video-JS. I believe it was
something like HTML5 VIDEO-JS GOOGLE DFP PRE ROLL JS Setup.

I’ve implemented the code in multiple variations.

Personally, I favor separating VideoJS JSON options as an array variable
for options which get passed to the player JS code where the DFP PREROLL
ADDON was included.

Then is the video HTML and CSS.

I found more problems related to styling CSS responsively for the video.

The next issue was solving coding the video player player to run the AD
before the video.

I hope this helps. If you have issues, send your code back and I will
review it.
